### PR TITLE
Feature/hotel listing reservations

### LINKED ISSUE #132: Listing reservations on hotel side
- [x] Reservation DataAccess
- [x] Reservation Service
- [x] Reservation Controller 
- [x] Integration tests for DataAccess
- [x] Unit tests for Service

Time spent: 2,5 hours

### PR DESCRIPTION
**Pull request by Michalski-Artur**
Closes #132 
Listing reservations on hotel side:
- [x] ReservationDataAccess
- [x] ReservationService
- [x] ReservationController
- [x] DataAccess tests
- [x] Service tests

### CODE REVIEW

File: Server-Module/Server/Services/Hotel/ReservationService/ReservationService.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    return result;
+            }
+
+            _transaction.BeginTransaction();
+            List<ReservationObjectView> reservationObjects = _dataAccess.GetReservations(hotelID, roomID, currentOnly, paging);
+            foreach(ReservationObjectView reservationObject in reservationObjects)
+            {
+                reservationObject.Client = _dataAccess.GetClientDetails(reservationObject.Client.ClientID);
+                reservationObject.Room = _dataAccess.GetRoomDetails(reservationObject.Room.RoomID);
+            }
```
Comment (daveedoo) [inline Server-Module/Server/Services/Hotel/ReservationService/ReservationService.cs:45]: Why is this here, not directly in `ReservationDataAcces`?

File: Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Server.Database.DataAccess.Hotel
+{
+    public interface IReservationDataAccess
+    {
+        ClientView GetClientDetails(int clientID);
+        RoomView GetRoomDetails(int roomID);
```
Comment (daveedoo) [inline Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs:13]: Are these methods part of `IReservationDataAccess` responsibility on purpose? Shouldn't they be more like private methods maybe?

**Review by daveedoo — COMMENTED** [review]
Good job. But what about the hotel module?

File: Server-Module/Server/Services/Hotel/ReservationService/ReservationService.cs (hunk 1 above)
Comment (Michalski-Artur) [reply Server-Module/Server/Services/Hotel/ReservationService/ReservationService.cs:45]: Yeah, I also have mixed feelings about it. Moving to DataAccess :)

**Comment by Michalski-Artur** [comment]
I moved GetClientDetails and GetRoomDetails entirely to ReservationDataAccess. Thanks for suggestion @daveedoo. I would appreciate if you would take a glance at it :)

File: Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs [hunk 3]
```diff
@@ -9,8 +9,6 @@ namespace Server.Database.DataAccess.Hotel
 {
     public interface IReservationDataAccess
     {
-        ClientView GetClientDetails(int clientID);
-        RoomView GetRoomDetails(int roomID);
         List<ReservationObjectView> GetReservations(int hotelID, int? roomID, bool? currentOnly, Paging paging);
         int? FindRoomAndGetOwner(int roomID);
```
Comment (daveedoo) [inline Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs:13]: .

**Review by daveedoo — COMMENTED** [review]
One more thing I missed previously - I don't see if it's used anywhere.
Cool though! :+1:

File: Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs (hunk 3 above)
Comment (Michalski-Artur) [reply Server-Module/Server/Database/DataAccess/Hotel/ReservationsManagement/IReservationDataAccess.cs:13]: It is used in Service to check if room exists and hotel demanding reservations is its owner

**Review by daveedoo — APPROVED** [review]
